### PR TITLE
fix: Improve error message on oras copy.go with  non-existent tags

### LINKED ISSUE #791: Improve error message on `oras copy` with non-existent tags
### What happened in your environment?

Error message is not clear when the tag `3.9` does not exist in the `alpine` folder.

```console
$ oras cp --from-oci-layout alpine:3.9 localhost:5000/alpine
Error: not found
```

The `not found` is correct but it is hard to let the users know whether the folder does not exist or the tag does not exist.

I do notice the error message was improved in [oras-go](https://github.com/oras-project/oras-go/blob/v2.0.0/copy.go#L138) but the following code is not improved:

https://github.com/oras-project/oras/blob/586fa838f00ea406696a29a571a90fab3f5a1cca/cmd/oras/cp.go#L139-L149

### What did you expect to happen?

Print something like
```
Error: failed to resolve alpine:3.9: not found
```

### How can we reproduce it?

```sh
oras cp docker.io/library/alpine:3.9 --to-oci-layout alpine
oras cp --from-oci-layout alpine:3.9 localhost:5000/alpine
```

### What is the version of your ORAS CLI?

Version:        1.0.0-rc.1
Go version:     go1.19.5
Git commit:     a98931bc68a0e7a5de6c51df8e5aa09aadad3057
Git tree state: clean

### What is your OS environment?

Ubuntu 20.04

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

### PR DESCRIPTION
**Pull request by khanjasir90**
**What this PR does / why we need it**:
Improves Error message on oras copy.go with non-existent tags
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #791 

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by khanjasir90** [comment]
@shizhMSFT @qweeah could u pls review it!

File: cmd/oras/cp.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ func runCopy(opts copyOptions) error {
 	if dstRef := opts.To.Reference; dstRef == "" {
 		desc, err = oras.Resolve(ctx, src, opts.From.Reference, rOpts)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to resolve %s: %w", src, err)
```
Comment (qweeah) [inline cmd/oras/cp.go:142]:
```suggestion
			return fmt.Errorf("failed to resolve %s: %w", opts.From.Reference, err)
```
A new ` oras.Resolve` call is added in line 154, better to copy the solution there also.

**Review by qweeah — CHANGES_REQUESTED** [review]

File: cmd/oras/cp.go (hunk 1 above)
Comment (shizhMSFT) [inline cmd/oras/cp.go:142]: `src` is not a string type

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

**Comment by khanjasir90** [comment]
@qweeah and @shizhMSFT I have made necessary give changes :)

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**